app/gui/src/dashboard/components/AriaComponents/ProgressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,44 @@
+/** @file Progress bar. */
+import {
+  ProgressBar as AriaProgressBar,
+  type ProgressBarProps as AriaProgressBarProps,
+} from '#/components/aria'
+import { tv } from '#/utilities/tailwindVariants'
+
+const PROGRESS_BAR_STYLES = tv({
+  base: 'h-2 rounded-full bg-primary/10',
+  slots: {
+    progressBar: 'h-full overflow-clip rounded-full bg-accent transition-width duration-1000',
+    indeterminateProgressBar: 'animate-horizontal-loader-1/6 h-full w-1/6 bg-white/30',
+  },
+})
+
+/** Props for a {@link ProgressBar}. */
+export interface ProgressBarProps extends Omit<AriaProgressBarProps, 'value'> {
+  readonly progress: number | 'indeterminate'
+}
+
+/** Progress bar. */
+export function ProgressBar(props: ProgressBarProps) {
```

Comment:
Please add stories

Reply:
added, lmk if there are more cases i should add and/or adjustments i should make to the component